### PR TITLE
JSON Substitution bug fixes

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

**Comment by vincentdass** [comment]
#6109 
#4884

File: Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts [hunk 1]
```diff
@@ -64,6 +64,66 @@ export function substituteJsonVariable(jsonObject, envObject) {
...
+    var multilineComment = 2;
+
+    for (var i = 0; i < content.length; i++) {
+        currentChar = content[i];
+        nextChar = i + 1 < content.length ? content[i + 1] : "";
+
+        if (insideComment) {
+            var update = false;
+            if (insideComment == singlelineComment && (currentChar + nextChar === '\r\n' || currentChar === '\n')) {
+                i--;
```
Comment (kmkumaran) [inline Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts]: Should you move forward?

File: Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts [hunk 2]
```diff
@@ -64,6 +64,66 @@ export function substituteJsonVariable(jsonObject, envObject) {
     }
 }
 
+function stripJsonComments(content) {
```
Comment (kmkumaran) [inline Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts]: Add L0, with single line, multi-line and some huge comments.

Comment (vincentdass) [reply Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts]: sure. planned to add them once I resolve the comments.

File: Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts (hunk 1 above)
Comment (vincentdass) [reply Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts]: This iterator(i) movement here just honors the new line coming after single line comment.
We should be good here, despite iterator backward movement (i--).

File: Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts (hunk 2 above)
Comment (vincentdass) [reply Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts]: Added L0 Tests

**Pull request merged**